xml/xen_administrating.xml

```diff
@@ -236,6 +237,77 @@
    </listitem>
   </itemizedlist>
 
+  <sect2 xml:id="sec-xen-manage-migrate-cpu">
+   <title>Detecting CPU Features</title>
+   <para>
+    By using the <systemitem>cpuid</systemitem> and
+    <systemitem>xen_maskcalc.py</systemitem> tools, you can compare features of
+    a CPU on the host from where you are migrating the source &vmguest; with
+    the features of CPUs on the target hosts. This way you can better predict
+    if the guest migrations will be successful.
+   </para>
+   <procedure>
+    <step>
+     <para>
+      Run the <command>cpuid -1r</command> command on each &dom0; that is
+      supposed to run or receive the migrated &vmguest; and capture the output
+      in text files, for example:
+     </para>
+<screen>
+<prompt>tux@vm_host1 &gt; </prompt>sudo cpuid -1r > vm_host1.txt
+<prompt>tux@vm_host2 &gt; </prompt>sudo cpuid -1r > vm_host2.txt
+<prompt>tux@vm_host3 &gt; </prompt>sudo cpuid -1r > vm_host3.txt
+</screen>
+    </step>
+    <step>
+     <para>
+      Copy all the output text files on a host with the
+      <command>xen_maskcalc.py</command> script installed.
+     </para>
+    </step>
+    <step>
+     <para>
+      Run the <command>xen_maskcalc.py</command> script on all output text
+      files:
+     </para>
+<screen>
+&prompt.sudo;xen_maskcalc.py vm_host1.txt vm_host2.txt vm_host3.txt
+cpuid = [
+    "0x00000001:ecx=x00xxxxxx0xxxxxxxxx00xxxxxxxxxxx",
+    "0x00000007,0x00:ebx=xxxxxxxxxxxxxxxxxx00x0000x0x0x00"
+]
+</screen>
+    </step>
+    <step>
+     <para>
+      Copy the output <literal>cpuid=[...]</literal> configuration snipped into
+      the <command>xl</command> configuration of the migrated
+      guest<filename>domU.cfg</filename> or alternatively to its &libvirt;'s
+      XML configuration.
```

Comment:
I'm not familiar with this tool, but is it really capable of producing libvirt <cpu> config from cpuid=[...] in xl.cfg? I guess we'll need some input from Olaf Hering.

Reply:
Yes, it produces also XML snippets. 

https://github.com/olafhering/xen_maskcalc/blob/9f48802ef81dd2370036e97897ab19b9cb15ca68/xen_maskcalc.py#L262